src/component/cyfs-group-lib/src/requestor.rs

```diff
@@ -0,0 +1,206 @@
+use std::sync::Arc;
+
+use cyfs_base::{
+    BuckyError, BuckyResult, JsonCodec, NamedObject, ObjectDesc, ObjectId, RawConvertTo,
+    CYFS_API_LEVEL,
+};
+use cyfs_core::{GroupProposal, GroupProposalObject};
+use cyfs_lib::{
+    HttpRequestorRef, NONAction, NONObjectInfo, NONOutputRequestCommon, NONRequestorHelper,
+    RequestorHelper,
+};
+use http_types::{Method, Request, Url};
+
+use crate::{
+    output_request::GroupStartServiceOutputRequest,
+    processor::{GroupOutputProcessor, GroupOutputProcessorRef},
+    GroupPushProposalOutputResponse, GroupStartServiceOutputResponse,
+};
+
+#[derive(Clone)]
+pub struct GroupRequestor {
+    dec_id: ObjectId,
+    requestor: HttpRequestorRef,
+    service_url: Url,
+}
+
+impl GroupRequestor {
+    pub fn new(dec_id: ObjectId, requestor: HttpRequestorRef) -> Self {
+        let addr = requestor.remote_addr();
+
+        let url = format!("http://{}/group/", addr);
+        let url = Url::parse(&url).unwrap();
+
+        Self {
+            dec_id,
+            requestor,
+            service_url: url,
+        }
+    }
+
+    pub fn clone_processor(&self) -> GroupOutputProcessorRef {
+        Arc::new(Box::new(self.clone()))
+    }
+
+    fn encode_common_headers(&self, com_req: &NONOutputRequestCommon, http_req: &mut Request) {
+        let dec_id = com_req.dec_id.as_ref().unwrap_or(&self.dec_id);
+        http_req.insert_header(cyfs_base::CYFS_DEC_ID, dec_id.to_string());
+
+        RequestorHelper::encode_opt_header_with_encoding(
+            http_req,
+            cyfs_base::CYFS_REQ_PATH,
+            com_req.req_path.as_deref(),
+        );
+
+        http_req.insert_header(CYFS_API_LEVEL, com_req.level.to_string());
```

Comment:
api level is non/ndn relatively obsolete design, for the group this function this should not be necessary